ext/srt/gstsrtobject.c

```
@@ -1531,12 +1531,16 @@ gst_srt_object_write_one (GstSRTObject * srtobject,
       case SRTS_BROKEN:
       case SRTS_NONEXIST:
       case SRTS_CLOSED:
-        GST_WARNING_OBJECT (srtobject->element,
-            "Invalid SRT socket. Trying to reconnect");
+        GST_ELEMENT_WARNING (srtobject->element, RESOURCE, WRITE, NULL,
+            ("Invalid SRT socket. Trying to reconnect. (%s)",
+                srt_getlasterror_str ()));
         gst_srt_object_close (srtobject);
         if (!gst_srt_object_open_internal (srtobject, cancellable, error)) {
           return -1;
         }
+        /* We can't send into a broken SRT socket. This will drop the current
```

I'm not sure if silently dropping buffers while we attempt to repair the broken connection is the best thing to do. Perhaps we should cease trying to reconnect, treat invalid socket as an unrecoverable state, return `GST_FLOW_ERROR` from `gst_srt_sink_render()` and stop streaming?

---

I guess the answer from upstream would be like; let user chose the behaviour when an srt socket is broken by a property.

For example, if `auto-reconnect` property is true, it is permanently trying to reconnect with silently dropping. Otherwise, when the socket is broken, it returns `GST_FLOW_ERROR` and stops streaming.

In my opinion, it's enough to drop silently when the socket is broken, then when a new _GstBuffer_ is coming, just trying to re-connect.